pkg/apis/serving/fieldmask.go

```diff
@@ -464,3 +464,28 @@ func SecurityContextMask(in *corev1.SecurityContext) *corev1.SecurityContext {
 
 	return out
 }
+
+// NamespacedObjectReferenceMask performs a _shallow_ copy of the Kubernetes ObjectReference
+// object to a new Kubernetes ObjectReference object bringing over only the fields allowed in
+// the Knative API. This does not validate the contents or the bounds of the provided fields.
+func NamespacedObjectReferenceMask(in *corev1.ObjectReference) *corev1.ObjectReference {
+	if in == nil {
+		return nil
+	}
+
+	out := new(corev1.ObjectReference)
+
+	// Allowed fields
+	out.APIVersion = in.APIVersion
+	out.Kind = in.Kind
+	out.Name = in.Name
+
+	// Disallowed
+	// This list is unnecessary, but added here for clarity
```

Comment:
/rant we do it everywhere, but it's bound to rot and mismatch :|

Reply:
Yeah, I did it for consistency.  We can talk to @dgerd when he's back.